aspnetcore/fundamentals/startup.md

```diff
@@ -1,75 +1,119 @@
+
+`IStartupFilter` implements a single method, [Configure](/dotnet/api/microsoft.aspnetcore.hosting.istartupfilter.configure), which receives and returns an `Action<IApplicationBuilder>`. An [IApplicationBuilder](/dotnet/api/microsoft.aspnetcore.builder.iapplicationbuilder) defines a class to configure an app's request pipeline. For more information, see [Creating a middleware pipeline with IApplicationBuilder](xref:fundamentals/middleware#creating-a-middleware-pipeline-with-iapplicationbuilder).
+
+Each `IStartupFilter` implements one or more middlewares in the request pipeline. The filters are invoked in the order they were added to the service container. Filters may add middleware before or after passing control to the next filter, thus they append to the beginning or end of the app pipeline.
+
+The [sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/startup/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample)) demonstrates how to register a middleware with `IStartupFilter`. The sample app includes a middleware that sets an options value from a query string parameter:
+
+[!code-csharp[Main](startup/sample/RequestSetOptionsMiddleware.cs?name=snippet1)]
+
+The `RequestSetOptionsMiddleware` is configured in the `RequestSetOptionsStartupFilter` class:
+
+[!code-csharp[Main](startup/sample/RequestSetOptionsStartupFilter.cs?name=snippet1&highlight=7)]
 
-The `Configure` method is used to specify how the ASP.NET application will respond to HTTP requests. The request pipeline is configured by adding [middleware](middleware.md) components to an `IApplicationBuilder` instance that is provided by dependency injection.
+The `IStartupFilter` is registered in the service container in `ConfigureServices`:
 
-In the following example from the default web site template, several extension methods are used to configure the pipeline with support for [BrowserLink](http://vswebessentials.com/features/browserlink), error pages, static files, ASP.NET MVC, and Identity.
+[!code-csharp[Main](startup/sample/Startup.cs?name=snippet1&highlight=3)]
 
-[!code-csharp[Main](../common/samples/WebApplication1/Startup.cs?highlight=8,9,10,14,17,19,21&start=58&end=84)]
+When a query string parameter for `option` is provided, the middleware processes the value assignment before the MVC middleware renders the response:
 
-Each `Use` extension method adds a [middleware](xref:fundamentals/middleware) component to the request pipeline. For instance, the `UseMvc` extension method adds the [routing](routing.md) middleware to the request pipeline and configures [MVC](xref:mvc/overview) as the default handler.
+![Browser window showing the rendered Index page. The value of Option is rendered as 'From Middleware' based on requesting the page with the query string parameter and value of option set to 'From Middleware'.](startup/_static/index.png)
```

Comment:
Per "MVC" guidelines, "images should be from a Safari or Chrome browser, unless the content is addressing core Windows technology (then writers should use Edge)".

Reply:
I didn't get a copy of those guidelines ... just the templates. May I have a copy so I can see the other guidelines?

Just curious: Any reason why? In the past the guidance was the other way around: @Rick-Anderson and @mairaw enforced a rule for Edge over the others.

Reply:
@guardrex I have emailed you a link to the internal doc.

Reply:
Awesome ... thx.

Reply:
No me. I never use edge.

Reply:
Oh ... my bad. I thought it was a PR ... Chrome shots were changed to Edge shots ... or maybe it was just that Chrome shots were there and Edge shots were swapped in. Anyway ... a little strange for MS not to promote its own tech. Edge is a fine browser.